src/databricks/labs/ucx/workspace_access/redash.py

```diff
@@ -83,8 +84,21 @@ def _applier_task(self, object_type: sql.ObjectTypePlural, object_id: str, acl:
         Please note that we only have SET option (DBSQL Permissions API doesn't support UPDATE operation).
         This affects the way how we prepare the new ACL request.
         """
-        self._ws.dbsql_permissions.set(object_type=object_type, object_id=object_id, access_control_list=acl)
-        return True
+        for _i in range(0, 3):
+            self._ws.dbsql_permissions.set(object_type=object_type, object_id=object_id, access_control_list=acl)
+
+            remote_permission = self._safe_get_dbsql_permissions(object_type, object_id)
+            if all(elem in remote_permission.access_control_list for elem in acl):
```

Comment:
Are you sure it's not required to convert from response to request?

Reply:
remote_permission.access_control_list is type List[sql.AccessControl] and acl is also list[sql.AccessControl]